### PR TITLE
Fix building the runtime from Windows

### PR DESCRIPTION
**Pull request by hughbe**
- And various unreachable warnings
- And change _MSC_VER conditions to _WIN32

### CODE REVIEW

File: stdlib/public/runtime/ImageInspectionWin32.cpp [hunk 1]
```diff
@@ -200,7 +200,7 @@ void swift::initializeProtocolConformanceLookup() {
   // rdar://problem/19045112
   const InspectArgs ProtocolConformancesArgs = {
     addImageProtocolConformanceBlockCallback,
-    ProtocolConformancesSection,
+    ProtocolConformancesSection
```
Comment (compnerd) [inline stdlib/public/runtime/ImageInspectionWin32.cpp]: Why the removal of the trailing comma?

File: stdlib/public/runtime/ImageInspectionWin32.cpp [hunk 2]
```diff
@@ -212,7 +212,7 @@ void swift::initializeTypeMetadataRecordLookup() {
   // rdar://problem/19045112
   const InspectArgs TypeMetadataRecordsArgs = {
     addImageTypeMetadataRecordBlockCallback,
-    TypeMetadataRecordsSection,
+    TypeMetadataRecordsSection
```
Comment (compnerd) [inline stdlib/public/runtime/ImageInspectionWin32.cpp]: Ditto.

**Comment by compnerd** [comment]
@swift-ci please test

File: stdlib/public/runtime/ImageInspectionWin32.cpp (hunk 1 above)
Comment (compnerd) [inline stdlib/public/runtime/ImageInspectionWin32.cpp]: I think that keeping that is better (I like the trailing commas, and it keeps the change focused).

Comment (hughbe) [inline stdlib/public/runtime/ImageInspectionWin32.cpp]: Fixed

File: stdlib/public/runtime/ImageInspectionWin32.cpp (hunk 2 above)
Comment (hughbe) [inline stdlib/public/runtime/ImageInspectionWin32.cpp]: Fixed - could you trigger the tests (or test+merge)

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Comment by hughbe** [comment]
Tests seem to have passed but this never merged. Might be worth checking to make sure! CI does lie occasionally

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Pull request merged**